src/services/services.ts

```diff
@@ -1223,6 +1231,31 @@ namespace ts {
             return sourceFile;
         }
 
+        function synchronizeAutoImportProvider(hostCache?: HostCache) {
+            Debug.assert(!syntaxOnly);
+
+            const dependencySelection = host.includePackageJsonAutoImports?.();

```

Comment:
Another thought is that given that project is sufficient to handle the creation of program, you want something like host.getAutoImportProviderHost which is subset of LanguageServiceHost to create a program.. You can then use that to create the program so you are ensuring not using same host. I have not thought about this enough but it seems like you want to get AuxillaryProject here as a host and createProgram instead.. THat way someone is not subscribing to breaking behaviour with getScriptSnapshot and similar)